packages/astro/src/core/build/plugins/plugin-ssr.ts

```diff
@@ -49,6 +52,9 @@ function vitePluginSSR(
 			if (id === SSR_VIRTUAL_MODULE_ID) {
 				return RESOLVED_SSR_VIRTUAL_MODULE_ID;
 			}
+			if (id === ADAPTER_VIRTUAL_MODULE_ID) {
+				return RESOLVED_ADAPTER_VIRTUAL_MODULE_ID;
```

Comment:
Would it work to just return `adapter.serverEntrypoint` here?

Reply:
Oh, interesting idea, let me try.

Reply:
Doesn't work, rollup complains about it being external again.